ui/src/app/components/apps-panel.tsx

```diff
@@ -0,0 +1,413 @@
+  const {
+    workspace: { namespace, googleProject },
+  } = props;
+
+  const [status, setRuntimeStatus] = useRuntimeStatus(namespace, googleProject);
+
+  // transition states, so we don't have to wait for polling
+  const [pausing, setPausing] = useState(false);
+  const [resuming, setResuming] = useState(false);
+
+  // when polling updates the status value, clear our transition states
+  useEffect(() => {
+    if (pausing) {
+      setPausing(false);
+    }
+    if (resuming) {
+      setResuming(false);
+    }
+  }, [status]);
+
+  // transition from RUNNING to STOPPED, or STOPPED to RUNNING
+  const toggleRuntimeStatus = () =>
+    switchCase(
+      status,
+      [
+        RuntimeStatus.Running,
+        () => {
+          setPausing(true);
+          setRuntimeStatus(RuntimeStatusRequest.Stop);
+        },
+      ],
+      [
+        RuntimeStatus.Stopped,
+        () => {
+          setResuming(true);
+          setRuntimeStatus(RuntimeStatusRequest.Start);
+        },
+      ]
+    );
+
+  const [icon, buttonText, enabled] = cond(
+    [
+      pausing || status === RuntimeStatus.Stopping,
+      () => [faSyncAlt, 'Pausing', false],
+    ],
+    [
+      resuming || status === RuntimeStatus.Starting,
+      () => [faSyncAlt, 'Resuming', false],
+    ],
+    [status === RuntimeStatus.Stopped, () => [faPlay, 'Resume', true]],
+    [status === RuntimeStatus.Running, () => [faPause, 'Pause', true]],
+    // choose a (disabled) default to show for other states
+    () => [faPause, 'Pause', false]
+  );
+
+  return (
+    <Clickable
+      disabled={!enabled}
+      style={{ padding: '0.5em' }}
+      onClick={toggleRuntimeStatus}
+    >
+      <FlexColumn style={enabled ? styles.button : styles.disabledButton}>
+        <FontAwesomeIcon {...{ icon }} style={styles.buttonIcon} />
+        <div style={styles.buttonText}>{buttonText}</div>
+      </FlexColumn>
+    </Clickable>
+  );
+};
+
+const RuntimeOpenButton = (props: { workspace: Workspace }) => {
+  const { workspace } = props;
+
+  const [showModal, setShowModal] = useState(false);
+  const [notebookNameList, setNotebookNameList] = useState<string[]>([]);
+
+  useEffect(() => {
+    notebooksApi()
+      .getNoteBookList(workspace.namespace, workspace.id)
+      .then((nbl) =>
+        setNotebookNameList(nbl.map((fd) => dropNotebookFileSuffix(fd.name)))
+      );
+  }, []);
+
+  return (
+    <>
+      {showModal && (
+        <NewNotebookModal
+          {...{ workspace }}
+          existingNameList={notebookNameList}
+          onClose={() => setShowModal(false)}
+        />
+      )}
+      <Clickable
+        style={{ padding: '0.5em' }}
+        onClick={() => setShowModal(true)}
+      >
+        <FlexColumn style={styles.button}>
+          <FontAwesomeIcon icon={faPlusCircle} style={styles.buttonIcon} />
+          <div style={styles.buttonText}>Create New</div>
+        </FlexColumn>
+      </Clickable>
+    </>
+  );
+};
+
+const ExpandedApp = (props: {
```

Comment:
Interesting test!  Maybe I should disable the create button when creation is happening?

At launch we will only allow a single Jupyter app to run at once (as is true today).  Unclear if we will ever want to change this.  (but the Terra/Leo backend supports it, if we do)

However, it will be possible to run one app environment _of each type_ simultaneously.